tests/Integration/PrestaShopBundle/Controller/Sell/Customer/Address/AddressControllerTest.php

```diff
@@ -0,0 +1,168 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\PrestaShopBundle\Controller\Sell\Customer\Address;
+
+use Country;
+use PrestaShop\PrestaShop\Core\Exception\TypeException;
+use Tests\Integration\PrestaShopBundle\Controller\GridControllerTestCase;
+use Tests\Integration\PrestaShopBundle\Controller\TestEntityDTO;
+
+class AddressControllerTest extends GridControllerTestCase
+{
+    private $countryId;
+
+    public function __construct($name = null, array $data = [], $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+
+        $this->gridRoute = 'admin_addresses_index';
+        $this->testEntityName = 'address';
+        $this->deleteEntityRoute = 'admin_addresses_delete';
+    }
+
+    /**
+     * Tests all provided entity filters
+     * All filters are tested in one test make tests run faster
+     *
+     * @throws TypeException
+     */
+    public function testAddressFilters(): void
+    {
+        foreach ($this->getTestFilters() as $testFilter) {
+            $this->assertFiltersFindOnlyTestEntity($testFilter);
+        }
+    }
+
+    /**
+     * @return array
+     */
+    protected function getTestFilters(): array
+    {
+        return [
+            ['address[id_address]' => $this->getTestEntity()->getId()],
+            ['address[firstname]' => 'firstname'],
+            ['address[lastname]' => 'testLa'],
+            ['address[address1]' => 'address1'],
+            ['address[postcode]' => '11111'],
+            ['address[city]' => 'stcity'],
+            ['address[id_country]' => $this->countryId],
+        ];
+    }
+
+    /**
+     * @return TestEntityDTO
+     */
+    protected function getTestEntity(): TestEntityDTO
+    {
+        return new TestAddressDTO(
+            $this->testEntityId,
+            'testfirstname',
+            'testlastname',
+            'testaddress1',
+            '11111',
+            'testcity',
+            'lithuania'
+        );
+    }
+
+    /**
+     * @return void
+     */
+    protected function createTestEntity(): void
```

Comment:
I have the feeling that this method could be more generic so that we don't need to re code it all for each new controller tested, basically the idea to get the `create` page and fill the form can be generalized and moved into the abstract class

The main complexity to make this function generic is the part where you override the address form data handler, since each page has its own `FormDataHandler` it will force us to override each one of them Instead we could override `PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Handler\FormHandler` which integrates every single for data handler And, like you did here, we can save the result in the `handleForm` method it returns a `FormHandlerResult` object which should contain the ID

This way we only have to perform one override in this PR, and the following one can just rely on the abstract class

Reply:
I thought I was going a bit overboard already with abstractions :D 

My biggest worry was that some forms may have more complex entity creation logic, but I guess you can always override this function for that specific test. So thanks for the feedback! Will implement it.